airbyte-cdk/bulk/toolkits/load-s3/src/main/kotlin/io/airbyte/cdk/load/file/s3/S3Client.kt

```diff
@@ -174,36 +178,67 @@ class S3Client(
 
 @Factory
 class S3ClientFactory(
+    private val arnRole: AWSArnRoleConfigurationProvider,
     private val keyConfig: AWSAccessKeyConfigurationProvider,
     private val bucketConfig: S3BucketConfigurationProvider,
-    private val uploadConifg: ObjectStorageUploadConfigurationProvider? = null,
+    private val uploadConfig: ObjectStorageUploadConfigurationProvider? = null,
 ) {
     companion object {
         fun <T> make(config: T) where
         T : S3BucketConfigurationProvider,
         T : AWSAccessKeyConfigurationProvider,
+        T : AWSArnRoleConfigurationProvider,
         T : ObjectStorageUploadConfigurationProvider =
-            S3ClientFactory(config, config, config).make()
+            S3ClientFactory(config, config, config, config).make()
     }
 
+    private val AIRBYTE_STS_SESSION_NAME = "airbyte-sts-session"
+    private val EXTERNAL_ID = "AWS_ASSUME_ROLE_EXTERNAL_ID"
+    private val AWS_ACCESS_KEY_ID = "AWS_ACCESS_KEY_ID"
+    private val AWS_SECRET_ACCESS_KEY = "AWS_SECRET_ACCESS_KEY"
+
     @Singleton
     @Secondary
     fun make(): S3Client {
-        val credentials = StaticCredentialsProvider {
-            accessKeyId = keyConfig.awsAccessKeyConfiguration.accessKeyId
-            secretAccessKey = keyConfig.awsAccessKeyConfiguration.secretAccessKey
-        }
 
-        val client =
+        val credsProvider: CredentialsProvider =
+            if (keyConfig.awsAccessKeyConfiguration.accessKeyId != null) {
+                StaticCredentialsProvider {
+                    accessKeyId = keyConfig.awsAccessKeyConfiguration.accessKeyId
+                    secretAccessKey = keyConfig.awsAccessKeyConfiguration.secretAccessKey
+                }
+            } else if (arnRole.awsArnRoleConfiguration.roleArn != null) {
+                // The Platform is expected to inject via credentials if ROLE_ARN is present.
+                val externalId = System.getenv(EXTERNAL_ID) // Consider injecting this dependency
+                val assumeRoleParams =
+                    AssumeRoleParameters(
+                        roleArn = arnRole.awsArnRoleConfiguration.roleArn!!,
+                        roleSessionName = AIRBYTE_STS_SESSION_NAME,
+                        externalId = externalId
+                    )
+                val creds = StaticCredentialsProvider {
+                    accessKeyId = System.getenv(AWS_ACCESS_KEY_ID)
+                    secretAccessKey = System.getenv(AWS_SECRET_ACCESS_KEY)
```

Comment:
These should be injected via micronaut.yaml file. I'm thinking it's simpler to do this in a follow up PR since this will probably result in some changes to the testing layer to inject env vars.